### PR TITLE
chore: fix Schematics for NG versions above 10.1

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a brief summary of this pull request.
There were some old methods, that are not supported by angular cli above and equal 10.1 version.
Now some of them are imported from `@schematics/angular` library.

To test it:
- You need testing app with ng@10.1 or above
- Fetch this branch
- run `npm run build-pack-library`
- copy `.tgz` file from `dist/libs/core/` to testing app directory
- Add this version of lib by writing `ng add @fundamental-ngx/core@{{path to tgz file}}`

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by InnaAtanasova — APPROVED** [review]

File: libs/core/schematics/ng-add/index.ts [hunk 1]
```diff
@@ -2,13 +2,13 @@ import { Rule, SchematicContext, Tree, chain, SchematicsException } from '@angul
...
 import { addPackageJsonDependency, NodeDependency, NodeDependencyType } from '@schematics/angular/utility/dependencies';
-import { addImportToRootModule, getProjectFromWorkspace, hasModuleImport } from '../utils/ng-module-utils';
-import { getAppModulePath } from '@schematics/angular/utility/ng-ast-utils';
-import { getWorkspace } from '@schematics/angular/utility/config';
+import { hasModuleImport } from '../utils/ng-module-utils';
 import { WorkspaceSchema } from '@schematics/angular/utility/workspace-models';
 
 import { cdkVersion } from './versions';
 import { defaultFontStyle } from './styles';
+import { addModuleImportToModule, getProjectFromWorkspace, getProjectMainFile, getAppModulePath } from '@angular/cdk/schematics';
```
Comment (fkolar) [inline libs/core/schematics/ng-add/index.ts:10]: How do you solve this CDK dependency. ? CDK needed be installed even before npm install kicks in. If you dont have CDK installed , it works?

Comment (JKMarkowski) [reply libs/core/schematics/ng-add/index.ts:10]: @fkolar  Good point, it throws error; `An unhandled exception occurred: Cannot find module '@angular/cdk/schematics'`. Anyway in `fundamental-ngx/core` package.json, CDK lib is added to peerDependency list, so it's mandatory. Should I take some methods straight from cdk then, instead of importing?

Comment (fkolar) [reply libs/core/schematics/ng-add/index.ts:10]: The cdk needs to be there before peerDependency ;-) But I am testing something here locally on some other project (blank empty), where I tried to do what you did, but somehow it reads CDK from somewhere. and it works. still not sure how its possible. 

Maybe cuz I am doing npm link? hmm.

**Comment by fkolar** [comment]
Just found interesting thing. as I kept meesign up wiht schematics and try to edit directly some node_module files to trace some problem such as ast-utils.js, I did nto reflected changes in host (target) project I did. But It used my other changes that I made in the source repo. 

So it looks like when you compile schematics it also compiles together nessesary libraries and functions. is this possible ?

**Comment by JKMarkowski** [comment]
@fkolar 
> Just found interesting thing. as I kept meesign up wiht schematics and try to edit directly some node_module files to trace some problem such as ast-utils.js, I did nto reflected changes in host (target) project I did. But It used my other changes that I made in the source repo.
> 
> So it looks like when you compile schematics it also compiles together nessesary libraries and functions. is this possible ? Or it explains why CDK works as well but when you dont have local reference it fails

So for me
- I created test app, without CDK/NGX
- Tried to add this package by `ng add @fundamental...`
- There was an error, that module from `@angular/cdk/schematics` couldn't be found, so it just relies on import and doesn't compile imported code, before deploy

**Comment by fkolar** [comment]
Then it looks like we need to rely on @schematics/angular functions instead of cdk

**Comment by fkolar** [comment]
@JKMarkowski 
I found this project that is also using CDK schematics. it has cdk in the dependencies of published package.json

https://github.com/NG-ZORRO/ng-zorro-antd

**Comment by JKMarkowski** [comment]
> @JKMarkowski
> I found this project that is also using CDK schematics. it has cdk in the dependencies of published package.json
> 
> https://github.com/NG-ZORRO/ng-zorro-antd
> 
> Before they start doing anything they first install depenencies I think

I see we are already adding CDK in schematics, but as you said, the order of methods in ngAdd chain has to be changed. I will test it

**Comment by JKMarkowski** [comment]
So now `@angular/cdk` is added as dependency, it will be installed with `ngx/core` if needed. Schematics are working correct now
